src/acl/ConnMark.cc

```diff
@@ -42,15 +42,21 @@ int
 Acl::ConnMark::match(ACLChecklist *cl)
 {
     const auto *checklist = Filled(cl);
-    const auto connmark = checklist->conn()->clientConnection->nfConnmark;
 
-    for (const auto &m : marks) {
-        if (m.matches(connmark)) {
-            debugs(28, 5, "found " << m << " matching " << asHex(connmark));
-            return 1;
+    if (checklist->conn() && checklist->conn()->clientConnection) {
+        const auto connmark = checklist->conn()->clientConnection->nfConnmark;
```

Comment:
The conn() method is fairly expensive and we are now using it thrice in a row for the common case. It would be nice to optimize this (and make this code more consistent across Squid):

```suggestion
    const auto conn = checklist->conn();
    if (conn && conn->clientConnection) {
        const auto connmark = conn->clientConnection->nfConnmark;
```

We can also reduce code duplication and the probability of a similar bug resurfacing by adding an ACLFilledChecklist::clientConnection() method.

I do not insist on these changes.